internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -214,6 +216,16 @@ func (r *BMACReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (c
 
 	agent := r.findAgent(ctx, bmh)
 
+	// Finalizer is only needed if the user has indicated that they want full management of the BMH
+	// this means the spoke node will be removed when the BMH is deleted
```

Comment:
Does "full management" means *only* "the spoke node will be removed when the BMH is deleted"? If so, it would probably be more clear to give the annotation a name that more closely matches the behavior it induces.

Reply:
I like that. Though that brings up another good point; we're deleting the Agent, not the Node, right?

Reply:
In this PR, but it's just the first bit that I got done. I have a branch that adds node deletion, but that's still in progress.

They will use the same annotation unless we think there's value in doing one and not the other in some case.

Reply:
`remove-agent-and-node-on-delete` it is then. :smile: Unless you have a better idea.